### PR TITLE
Added new command line options for logger.

### PR DESCRIPTION
**Pull request by akallabeth**
The new command line options /log-level:<level> and
/log-filters:<filter>:<level> allow setting default log
level and log module filter to be set.

### CODE REVIEW

**Comment by mfleisz** [comment]
+1

File: winpr/libwinpr/utils/wlog/wlog.c [hunk 1]
```diff
@@ -302,6 +305,92 @@ DWORD WLog_GetLogLevel(wLog* log)
...
+		p++;
+	}
+
+	pos = g_FilterCount;
+	size = g_FilterCount + count;
+	tmp = (wLogFilter*) realloc(g_Filters, size * sizeof(wLogFilter));
+
+	if (!tmp)
+	{
+		free (g_Filters);
```
Comment (hardening) [inline winpr/libwinpr/utils/wlog/wlog.c]: Do we need to free g_Filters ?

File: winpr/libwinpr/utils/wlog/wlog.c [hunk 2]
```diff
@@ -302,6 +305,92 @@ DWORD WLog_GetLogLevel(wLog* log)
...
+	cp = (LPSTR)_strdup(filter);
+	if (!cp)
+		return FALSE;
+
+	p = cp;
+	s = cp;
+
+	do
+	{
+		p = strchr(p, ',');
```
Comment (hardening) [inline winpr/libwinpr/utils/wlog/wlog.c]: isn't it 
p = strchr(s, ','); 
here ?

May I suggest renaming s to startAt and p to commaPos ?

File: winpr/libwinpr/utils/wlog/wlog.c [hunk 3]
```diff
@@ -302,6 +305,92 @@ DWORD WLog_GetLogLevel(wLog* log)
...
+				return FALSE;
+			}
+		}
+		else
+			break;
+
+		if (p)
+		{
+			s = p + 1;
+			p++;
```
Comment (hardening) [inline winpr/libwinpr/utils/wlog/wlog.c]: if you take my suggestion above, incrementing p is useless

**Comment by hardening** [comment]
Some non blocking nitpick remarks.
+1

File: winpr/libwinpr/utils/wlog/wlog.c (hunk 2 above)
Comment (akallabeth) [inline winpr/libwinpr/utils/wlog/wlog.c]: @hardening The line is correct (```s``` always points to the beginning of the last string, ```p``` to the starting point for the next search / last found ```,```) but I'll rename them.

File: winpr/libwinpr/utils/wlog/wlog.c (hunk 1 above)
Comment (akallabeth) [inline winpr/libwinpr/utils/wlog/wlog.c]: @hardening If realloc fails that is the way to go, but you pointed out something I missed. Need to reset all static globals.

Comment (akallabeth) [inline winpr/libwinpr/utils/wlog/wlog.c]: @hardening on second thought, maybe it is easier your way, just abort before update of the globals and let the uninit handle that...

**Comment by bmiklautz** [comment]
@akallabeth what does take preference now? Command line or environment?

**Comment by akallabeth** [comment]
@bmiklautz Command line for level, filters are merged.

**Pull request merged**